zetasql-toolkit-bigquery/src/main/java/com/google/zetasql/toolkit/catalog/bigquery/BigQueryCatalog.java

```diff
@@ -576,7 +601,14 @@ public void addAllTablesUsedInQuery(String query, AnalyzerOptions options) {
             .map(tablePath -> String.join(".", tablePath))
             .filter(BigQueryReference::isQualified) // Remove non-qualified tables
             .collect(Collectors.toSet());
-    this.addTables(ImmutableList.copyOf((tables)));
+    Set<String> standardTables =
+        tables.stream()
+            .filter(ref -> !BigQueryReference.isWildcardReference(ref))
+            .collect(Collectors.toSet());
+    Set<String> wildcardTables =
+        tables.stream().filter(BigQueryReference::isWildcardReference).collect(Collectors.toSet());
+    this.addTables(ImmutableList.copyOf((standardTables)));
+    this.addWildcardTables(ImmutableList.copyOf((wildcardTables)));
```

Comment:
The last bit of feedback before we merge.

I think it would be better if we keep `addAllTablesUsedInQuery()` as it was and move this separation between standard vs wildcard tables to the `addTables()` method.

That way people can do something like `catalog.addTable("bigquery-public-data.noaa_gsod.gsod202*")`.

Reply:
hi, i'm replying from my 2nd acc. let me work on it today, sorry for the delay.